SalesforceCommerceCloud/pwa-kit
# Re-structure all @chakra-ui imports to point imports to local `shared` dirs @W-13623534

By re-structuring to import and re-export chakra, we incur zero bundle bloat and allow customer to selectively remove certain chakra components that they don't use (or find objectionable in terms of bundle size) in favor of their own, more optimal components in a "mix and match" theme. Granted, we know that removing chakra altogether is a much bigger task, but this PR both enables that (though difficult) and **also** empowers developers to make better use of Template Extensibility to reduce bundlesize by targeting components with the `overrides` system and exporting different components that are more lightweight (bundle size) or more fit to purpose

In my testing, I changed the following files

<img width="559" alt="Screenshot 2023-06-22 at 4 44 38 PM" src="https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47546998/1ddb7814-f218-486f-96ae-61c0b2ada709">

And saw a bundle size reduction from `298kb`
<img width="876" alt="Screenshot 2023-06-22 at 4 40 36 PM" src="https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47546998/e46045af-304a-45f3-aac8-3bd74a4e79ba">


to `283kb` by removing just a few small components
<img width="864" alt="Screenshot 2023-06-22 at 4 44 27 PM" src="https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47546998/f9cda905-b5a6-47ef-9fdc-0aafc2089572">

Thus demonstrating and validating our plan that this strategy is viable as a less difficult path for customers that want to use our template and also reduce their bundlesize. in addition it demonstrates that chakra tree shaking works as expected, which is important in getting the benefit.



# Description



# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Every file that has an import of `@chakra-ui/theme` / `@chakra-ui/react` / or `@chakra-ui/skip-nav` is updated to point to the project-relative import

# How to Test-Drive This PR

- Check out the branch `feature/ui-component-restructure`
- run `npm ci` at the monorepo root
- `cd packages/template-retail-react-app`
- `npm start`
- verify that the project still functions without import errors

# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
